api-client/src/main/java/com/xing/api/RealCallSpec.java

```diff
@@ -231,9 +237,23 @@ public void cancel() {
         return this;
     }
 
+    /** Timeouts in seconds >0. */
+    @Override public CallSpec<RT, ET> timeouts(int connectTimeout, int readTimeout) {
+        this.connectTimeout = connectTimeout;
+        this.readTimeout = readTimeout;
+        return this;
+    }
+
     /** Returns a raw {@link Call} pre-building the targeted request. */
     private Call createRawCall() {
-        return api.client().newCall(builder.request());
+        OkHttpClient client = api.client();
+        if (readTimeout > 0 && connectTimeout > 0) {
+            client = client.newBuilder()
```

Comment:
I don't think so. If you do `.newBuilder()` it will pre-fill the new builder with all the defaults of the original instance, even so with the same `Dispatcher` instance that holds the `ExecutorService` on which the calls are processed.